docs/source/python/ipc.rst

```diff
@@ -105,6 +105,11 @@ An important point is that if the input source supports zero-copy reads
 (e.g. like a memory map, or ``pyarrow.BufferReader``), then the returned
 batches are also zero-copy and do not allocate any new memory on read.
 
+All stream objects have a ``close`` method. In the above example 
+:func:`~pyarrow.BufferOutputStream.getvalue` both closes and returns the buffer.
+If the return object is not needed, call the relevent ``close`` method
+after you are finished using the object to avoid data corruption issues.
```

Comment:
This is a little specific to BufferOutputStream (which is used for convenience in this example). I think the relevant point is that closing the writer does not close the stream, so you must close the stream/file yourself to ensure all data is persisted. (And maybe note that in this example, we don't do that as `getvalue` implicitly closes the stream.)

Reply:
Also mention that the context protocol should work as well (i.e. the `with` statement).